File: prototypes/research-assistant/typescript/src/tools/summarize.ts
export function summarize(text: string, maxLength = 200): string {
  if (text.length <= maxLength) return text;
  const truncated = text.slice(0, maxLength);
  const lastSpace = truncated.lastIndexOf(" ");
  return `${lastSpace > 0 ? truncated.slice(0, lastSpace) : truncated}...`;
}

File: prototypes/research-assistant/python/app/tools/summarize.py
"""Text summarization tool."""


async def summarize(text: str, max_length: int = 200) -> str:
    """Summarize text by truncating to max_length characters."""
    if len(text) <= max_length:
        return text
    return text[:max_length].rsplit(" ", 1)[0] + "..."
